packages/react-native/ReactCommon/react/renderer/components/view/ViewProps.cpp

```diff
@@ -412,6 +422,56 @@ BorderMetrics ViewProps::resolveBorderMetrics(
   };
 }
 
+Transform ViewProps::resolveTransform(
+  LayoutMetrics const &layoutMetrics) const  {
+  float viewWidth = layoutMetrics.frame.size.width;
+  float viewHeight = layoutMetrics.frame.size.height;
+  if (transformOrigin.empty() || (viewWidth == 0 && viewHeight == 0)) {
+    return transform;
+  }
+  auto newTransform = Transform{};
+  std::array<float, 3> translateOffsets = getTranslateForTransformOrigin(viewWidth, viewHeight);
+  newTransform = newTransform * Transform::Translate(translateOffsets[0], translateOffsets[1], translateOffsets[2]);
+  newTransform = newTransform * transform;
+  newTransform = newTransform * Transform::Translate(-translateOffsets[0], -translateOffsets[1], -translateOffsets[2]);
+  return newTransform;
+}
+
+// https://drafts.csswg.org/css-transforms/#transform-origin-property
+std::array<float, 3> ViewProps::getTranslateForTransformOrigin(float viewWidth, float viewHeight) const {
+  float viewCenterX = viewWidth / 2;
+  float viewCenterY = viewHeight / 2;
+
+  std::array<float, 3> origin = {viewCenterX, viewCenterY, 0.0f};
+  std::istringstream iss(transformOrigin);
+  std::string part;
+  for (int i = 0; std::getline(iss, part, ' ') && i < 3; i++) {
+    auto percentPos = part.find('%');
+    bool isPercent = percentPos != std::string::npos;
+    if (isPercent) {
+      origin[i] = (i == 0 ? viewWidth : viewHeight) * std::stof(part.substr(0, percentPos)) / 100.0f;
+    } else if (part == "top") {
+       origin[1] = 0.0f;
+    } else if (part == "bottom") {
+       origin[1] = static_cast<float>(viewHeight);
+    } else if (part == "left") {
+       origin[0] = 0.0f;
+    } else if (part == "right") {
+       origin[0] = static_cast<float>(viewWidth);
```

Comment:
This will currently accept the input "top left", which is not spec complaint. Do browsers accept?

Reply:
yeah, the `top left` [should work](https://codepen.io/nishanB/pen/VwVjyxO). It'll take `-viewWidth/2 , -viewHeight/2` as the offsets. It also works in any order in browser i.e. `top left`, `left top`. Invalid inputs could be `left left`, that results in `-viewWidth/2 0` offsets so it also works same as browsers. `left`, `right` affects horizontal offset and `top` ,`bottom` vertical and `center` is the initial value - [Spec](https://www.w3.org/TR/css-transforms-1/#transform-origin-property). So, `right` always take priority over `center` even if comes later in order e.g. `right center`, `center right` results in the same origin. I can surely add test cases! 
I am out this week as well. I'll pick this up as soon as I am back home! 😅